addons/headless/functions/fnc_transferGroups.sqf

```diff
@@ -125,98 +98,42 @@ private _numTransferredHC3 = 0;
-
-            if (_transferred) then {
-                _numTransferredHC2 = _numTransferredHC2 + 1;
-            };
+    // Don't transfer if it's already local to HC
+    if (_previousOwner != _idHC) then {
+        [QGVAR(groupTransferPre), [_x, _HC, _previousOwner, _idHC], _previousOwner] call CBA_fnc_ownerEvent; // API
+        [QGVAR(groupTransferPre), [_x, _HC, _previousOwner, _idHC], _idHC] call CBA_fnc_ownerEvent; // API
+
+        private _transferred = _x setGroupOwner _idHC;
+
+        [QGVAR(groupTransferPost), [_x, _HC, _previousOwner, _idHC, _transferred], _previousOwner] call CBA_fnc_ownerEvent; // API
+        [QGVAR(groupTransferPost), [_x, _HC, _previousOwner, _idHC, _transferred], _idHC] call CBA_fnc_ownerEvent; // API
+
+        if (_transferred) then {
+            _numTransferred set [_currentIndexHC, (_numTransferred select _currentIndexHC) + 1];
         };
-        case 3: {
-            if (_loadBalance) then {
-                // Find the next valid HC
-                // If none are valid, _currentHC will remain the same
-                if (_idHC1 != -1) then {
-                    _currentHC = 1;
-                } else {
-                    if (_idHC2 != -1) then {
-                        _currentHC = 2;
-                    };
-                };
-            };
-
-            // Don't transfer if it's already local to HC3
-            if (_previousOwner == _idHC3) exitWith {};
-
-            [QGVAR(groupTransferPre), [_x, _HC3, _previousOwner, _idHC3], _previousOwner] call CBA_fnc_ownerEvent; // API
-            [QGVAR(groupTransferPre), [_x, _HC3, _previousOwner, _idHC3], _idHC3] call CBA_fnc_ownerEvent; // API
-
-            private _transferred = _x setGroupOwner _idHC3;
-
-            [QGVAR(groupTransferPost), [_x, _HC3, _previousOwner, _idHC3, _transferred], _previousOwner] call CBA_fnc_ownerEvent; // API
-            [QGVAR(groupTransferPost), [_x, _HC3, _previousOwner, _idHC3, _transferred], _idHC3] call CBA_fnc_ownerEvent; // API
-
-            if (_transferred) then {
-                _numTransferredHC3 = _numTransferredHC3 + 1;
-            };
+    };
+
+    // Find the next valid HC if more than one HC is present
+    if (_loadBalance) then {
+        if (_currentIndexHC < _countHC) then {
+            _currentIndexHC = _currentIndexHC + 1;
+        } else {
+            _currentIndexHC = 0;
         };
     };
 } forEach allGroups;
 
 if (XGVAR(log)) then {
-    private _numTransferredTotal = _numTransferredHC1 + _numTransferredHC2 + _numTransferredHC3;
-    INFO_4("Groups Transferred: Total: %1 - HC1: %2 - HC2: %3 - HC3: %4",_numTransferredTotal,_numTransferredHC1,_numTransferredHC2,_numTransferredHC3);
+    private _numTransferredTotal = 0;
+
+    {
+        _numTransferredTotal = _numTransferredTotal + _x;
+    } forEach _numTransferred;
+
+    INFO_2("Groups Transferred: Total: %1 - per HC: %2",_numTransferredTotal,_numTransferred);
```

Comment:
Not sure if this is sufficient or if I should format the array into a cleaner string.